Microsoft.Toolkit.Uwp.UI.Controls/ImageEx/ImageEx.Source.cs

```diff
@@ -57,45 +60,54 @@ private static bool IsHttpUri(Uri uri)
         {
-            if (_isInitialized)
-            {
-                _image.Source = null;
+            await semaphore.WaitAsync();
 
-                if (source == null)
+            try
+            {
+                if (_isInitialized)
                 {
-                    VisualStateManager.GoToState(this, UnloadedState, true);
-                    return;
-                }
+                    _image.Source = null;
 
-                VisualStateManager.GoToState(this, LoadingState, true);
+                    if (source == null)
+                    {
+                        VisualStateManager.GoToState(this, UnloadedState, true);
+                        return;
+                    }
 
-                var imageSource = source as ImageSource;
-                if (imageSource != null)
-                {
-                    _image.Source = imageSource;
-                    ImageExOpened?.Invoke(this, new ImageExOpenedEventArgs());
-                    VisualStateManager.GoToState(this, LoadedState, true);
-                    return;
-                }
+                    VisualStateManager.GoToState(this, LoadingState, true);
 
-                _uri = source as Uri;
-                if (_uri == null)
-                {
-                    var url = source as string ?? source.ToString();
-                    if (!Uri.TryCreate(url, UriKind.RelativeOrAbsolute, out _uri))
+                    var imageSource = source as ImageSource;
+                    if (imageSource != null)
                     {
-                        VisualStateManager.GoToState(this, FailedState, true);
+                        _image.Source = imageSource;
+                        ImageExOpened?.Invoke(this, new ImageExOpenedEventArgs());
+                        VisualStateManager.GoToState(this, LoadedState, true);
                         return;
                     }
-                }
 
-                _isHttpSource = IsHttpUri(_uri);
-                if (!_isHttpSource && !_uri.IsAbsoluteUri)
-                {
-                    _uri = new Uri("ms-appx:///" + _uri.OriginalString.TrimStart('/'));
-                }
+                    _uri = source as Uri;
+                    if (_uri == null)
+                    {
+                        var url = source as string ?? source.ToString();
+                        if (!Uri.TryCreate(url, UriKind.RelativeOrAbsolute, out _uri))
+                        {
+                            VisualStateManager.GoToState(this, FailedState, true);
+                            return;
+                        }
+                    }
 
-                await LoadImageAsync();
+                    _isHttpSource = IsHttpUri(_uri);
+                    if (!_isHttpSource && !_uri.IsAbsoluteUri)
+                    {
+                        _uri = new Uri("ms-appx:///" + _uri.OriginalString.TrimStart('/'));
+                    }
+
+                    await LoadImageAsync();
```

Comment:
I think only this line has to use semaphore, since all other stuff is synchronous. I might also be wrong.